src/shogun/base/SGObject.cpp

```diff
@@ -1092,16 +1092,24 @@ void CSGObject::init_auto_params()
 	}
 }
 
+CSGObject* CSGObject::get(const std::string& name, index_t index) const
+{
+	if (auto* result = get_sgobject_type_dispatcher<CKernel>(this, name, index))
```

Comment:
The reason I'd like to unify is that I want to use the dispatching function here and in the standard get, with a different lambda that binds the index.

Reply:
Next question would be to avoid listing the base types twice